provider/akamai/akamai.go

```diff
@@ -168,215 +217,293 @@ func (p *AkamaiProvider) fetchZones() (zones akamaiZones, err error) {
-			continue
+// cleanTargets preps recordset rdata if necessary for EdgeDNS
+func cleanTargets(rtype string, targets ...string) []string {
+	log.Debugf("Targets to clean: [%v]", targets)
+	//var targets []string = tgts
+	if rtype == "CNAME" || rtype == "SRV" {
+		for idx, target := range targets {
+			targets[idx] = strings.TrimSuffix(target, ".")
+		}
+	} else if rtype == "TXT" {
+		for idx, target := range targets {
+			log.Debugf("TXT data to clean: [%s]", target)
+			// need to embed text data in quotes. Make sure not piling on
+			target = strings.Trim(target, "\"")
+			// bug in DNS API with embedded quotes.
+			if strings.Contains(target, "owner") && strings.Contains(target, "\"") {
+				target = strings.ReplaceAll(target, "\"", "`")
+			}
+			targets[idx] = "\"" + target + "\""
 		}
-		if zoneName, _ := zoneNameIDMapper.FindZone(endpoint.DNSName); zoneName != "" {
-			akamaiRecord := p.newAkamaiRecord(endpoint.DNSName, endpoint.RecordType, endpoint.Targets...)
-			body, _ := json.MarshalIndent(akamaiRecord, "", "  ")
+	}
+	log.Debugf("Clean targets: [%v]", targets)
 
-			log.Infof("Create new Endpoint at Akamai FastDNS - Zone: '%s', DNSName: '%s', RecordType: '%s', Targets: '%+v'", zoneName, endpoint.DNSName, endpoint.RecordType, endpoint.Targets)
+	return targets
+}
 
-			if p.dryRun {
-				continue
+// trimTxtRdata removes surrounding quotes for received TXT rdata
+func trimTxtRdata(rdata []string, rtype string) []string {
+	if rtype == "TXT" {
+		for idx, d := range rdata {
+			//rdata[idx] = strings.Trim(d, "\"")
+			if strings.Contains(d, "`") {
+				rdata[idx] = strings.ReplaceAll(d, "`", "\"")
 			}
-			_, err := p.request("POST", "config-dns/v2/zones/"+zoneName+"/names/"+endpoint.DNSName+"/types/"+endpoint.RecordType, bytes.NewReader(body))
-			if err != nil {
-				log.Errorf("Failed to create Akamai endpoint DNSName: '%s' RecordType: '%s' for zone: '%s'", endpoint.DNSName, endpoint.RecordType, zoneName)
-				failed = append(failed, endpoint)
-				continue
+		}
+	}
+	log.Debugf("Trimmed data: [%v]", rdata)
+
+	return rdata
+}
+
+func ttlAsInt(src endpoint.TTL) int {
+	var temp interface{} = int64(src)
+	var temp64 = temp.(int64)
+	var ttl int = edgeDNSRecordTTL // int
+	if temp64 > 0 && temp64 <= int64(maxInt) {
+		ttl = int(temp64)
+	}
+
+	return ttl
+}
+
+// Create Endpoint Recordsets
+func (p AkamaiProvider) createRecordsets(zoneNameIDMapper provider.ZoneIDName, endpoints []*endpoint.Endpoint) error {
+	if len(endpoints) == 0 {
+		log.Info("No endpoints to create")
+		return nil
+	}
+
+	endpointsByZone := edgeChangesByZone(zoneNameIDMapper, endpoints)
+
+	// create all recordsets by zone
+	for zone, endpoints := range endpointsByZone {
+		recordsets := &dns.Recordsets{Recordsets: make([]dns.Recordset, 0)}
+		for _, endpoint := range endpoints {
+			newrec := newAkamaiRecordset(endpoint.DNSName,
+				endpoint.RecordType,
+				ttlAsInt(endpoint.RecordTTL),
```

Comment:
why not simply use int(endpoint.RecordTTL) and drop the helper function?

Reply:
Type conversion is tricky. Underlying struct type is Int64. Also need check if > IntMax and adjust